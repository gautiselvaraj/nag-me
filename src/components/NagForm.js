import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Datetime from 'react-datetime';
import Input from './Input';
import Select from './Select';
import Button from './Button';
import Icon from './Icon';
import H3 from './H3';

const NagFormWrap = styled.form`
  padding: 0.5rem;
`;

const NagFormHeader = styled.div`
  align-items: center;
  background-color: ${props => props.theme.white};
  border-bottom: 1px solid ${props => props.theme.greyLighter};
  display: flex;
  padding: 5px;
  justify-content: space-between;
`;

const BackLink = styled.div`
  flex: 0 0 25px;
  height: 15px;
  text-align: center;
`;

const Heading = styled.div`
  flex-grow: 1;
  text-align: center;
`;

const Spacer = styled.div`
  margin-bottom: 0.75rem;
`;

const ButtonSpacer = styled.div`
  margin-bottom: 0.75rem;
  margin-top: 1.5rem;
`;

const nagRepeatOptions = {
  min: [1, 2, 3, 5, 10, 15, 20, 30, 45],
  hour: Array.from(Array(23)).map((_, i) => i + 1),
  day: Array.from(Array(6)).map((_, i) => i + 1),
  week: Array.from(Array(10)).map((_, i) => i + 1),
  month: Array.from(Array(11)).map((_, i) => i + 1),
  year: Array.from(Array(10)).map((_, i) => i + 1)
};

const today = Datetime.moment().subtract(1, 'd');

export default class NagForm extends Component {
  static propTypes = {
    nagIndex: PropTypes.func.isRequired,
    nagCreate: PropTypes.func.isRequired,
    nagUpdate: PropTypes.func.isRequired,
    editNagId: PropTypes.number,
    editNag: PropTypes.object
  };

  state = {
    title: this.props.editNag ? this.props.editNag.title : null,
    on: this.props.editNag ? this.props.editNag.nextNag : null,
    repeats: this.props.editNag ? this.props.editNag.repeats : null,
    titleError: null,
    onError: null
  };

  valid = current => current.isAfter(today);

  handleInputChange = e => {
    const target = e.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;

    this.setState({
      [name]: value,
      [`${name}Error`]: !value ? `Nag ${name} is required` : null
    });
  };

  handleDateTimeInputChange = moment => {
    this.setState({
      on: moment.valueOf(),
      onError: null
    });
  };

  handleSubmit = e => {
    e.preventDefault();
    const { title, on, repeats } = this.state;
    const { editNagId, nagUpdate, nagCreate } = this.props;

    if (!title || !on) {
      this.setState({
        titleError: !title ? 'Nag title is required' : null,
        onError: !on ? 'Nag on is required' : null
      });
      return false;
    }

    if (on < Date.now()) {
      this.setState({
        onError: 'Nag on should be in future'
      });
      return false;
    }

    if (editNagId) {
      nagUpdate(editNagId, {
        title,
        nextNag: on,
        repeats
      });
    } else {
      nagCreate({
        title,
        firstNag: on,
        nextNag: on,
        repeats
      });
    }
  };

  render() {
    const { nagIndex } = this.props;
    const { title, on, repeats, titleError, onError } = this.state;

    return (
      <div>
        <NagFormHeader>
          <BackLink>
            <Button reset title="Back" onClick={nagIndex}>
              <Icon back />
            </Button>
          </BackLink>
          <Heading>
            <H3>Create new Nag</H3>
          </Heading>
          <BackLink />
        </NagFormHeader>
        <NagFormWrap onSubmit={this.handleSubmit}>
          <Spacer>
            <Input
              autoFocus
              defaultValue={title}
              type="text"
              name="title"
              id="nag_title"
              label="What to Nag about?"
              error={titleError}
              onChange={this.handleInputChange}
            />
          </Spacer>
          <Spacer>
            <Datetime
              renderInput={props => (
                <Input
                  id="nag_on"
                  {...props}
                  readOnly
                  label="Nag on"
                  name="on"
                  error={onError}
                />
              )}
              dateFormat="MMM Do, YYYY"
              isValidDate={this.valid}
              onChange={this.handleDateTimeInputChange}
              defaultValue={
                on ? Datetime.moment(on).format('MMM Do, YYYY h:mm A') : ''
              }
            />
          </Spacer>
          <Spacer>
            <Select
              name="repeats"
              id="nag_repeats"
              label="Nag every"
              defaultValue={repeats}
              onChange={this.handleInputChange}
            >
              <option value="" />
              {Object.keys(nagRepeatOptions).map(key =>
                nagRepeatOptions[key].map(t => (
                  <option value={`${t} ${key}`} key={`${t} ${key}`}>
                    {t} {key}
                    {t === 1 ? '' : 's'}
                  </option>
                ))
              )}
            </Select>
          </Spacer>
          <ButtonSpacer>
            <Button block type="submit">
              Start Nagging
            </Button>
          </ButtonSpacer>
        </NagFormWrap>
      </div>
    );
  }
}
