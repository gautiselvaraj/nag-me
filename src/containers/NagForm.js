import {connect} from 'react-redux';
import NagForm from '../components/NagForm';
import {nagIndex, nagCreate, nagUpdate} from '../actions/NagActions';

const mapDispatchToProps = dispatch => ({
  nagIndex: () => dispatch(nagIndex()),
  nagCreate: nag => dispatch(nagCreate(nag)),
  nagUpdate: nag => dispatch(nagUpdate(nag)),
});

export default connect(null, mapDispatchToProps)(NagForm);
