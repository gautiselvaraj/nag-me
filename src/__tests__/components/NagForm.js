import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import moment from 'moment';
import MockDate from 'mockdate';
import NagForm from '../../components/NagForm';
import 'jest-styled-components';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureMockStore([]);

describe('<NagForm />', () => {
  let nagIndex, nagPause, nagCreate, nagUpdate;
  const nagObject = {
    id: 1,
    title: 'Test Nag 1',
    firstNag: 1234567890,
    nextNag: 1234567890,
    repeats: '1 min',
    status: 'LIVE',
    createdAt: 1234567890
  };

  beforeAll(() => {
    MockDate.set('10/10/2016');
  });

  afterAll(() => {
    MockDate.reset();
  });

  beforeEach(() => {
    nagIndex = jest.fn();
    nagCreate = jest.fn();
    nagUpdate = jest.fn();
  });

  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={mockStore()}>
          <NagForm
            nagIndex={nagIndex}
            nagCreate={nagCreate}
            nagUpdate={nagUpdate}
          />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have correct elements and children', () => {
    const nagForm = mount(
      <Provider store={mockStore()}>
        <NagForm
          nagIndex={nagIndex}
          nagCreate={nagCreate}
          nagUpdate={nagUpdate}
        />
      </Provider>
    );
    expect(nagForm.find('button[title="Back"]').length).toBe(1);
    expect(nagForm.find('h3').length).toBe(1);
    expect(nagForm.find('h3').text()).toBe('Create a nag');
    expect(nagForm.find('input#nag_title').length).toBe(1);
    expect(nagForm.find('input#nag_on').length).toBe(1);
    expect(nagForm.find('select#nag_repeats').length).toBe(1);
    expect(nagForm.find('button[type="submit"]').length).toBe(1);
  });

  it('should call nagIndex', () => {
    const nagForm = mount(
      <Provider store={mockStore()}>
        <NagForm
          nagIndex={nagIndex}
          nagCreate={nagCreate}
          nagUpdate={nagUpdate}
        />
      </Provider>
    );
    nagForm.find('button[title="Back"]').simulate('click');
    expect(nagIndex.mock.calls.length).toBe(1);
  });

  it('should call nagCreate when all fields are valid', () => {
    const nagForm = mount(
      <Provider store={mockStore()}>
        <NagForm
          nagIndex={nagIndex}
          nagCreate={nagCreate}
          nagUpdate={nagUpdate}
        />
      </Provider>
    );
    nagForm.find('form').simulate('submit');
    expect(nagCreate.mock.calls.length).toBe(0);

    const nagTitle = nagForm.find('input#nag_title');
    nagTitle.instance().value = 'Nag Title';
    nagTitle.simulate('change', nagTitle);
    const nagOn = nagForm.find('input#nag_on');
    nagOn.instance().value = moment()
      .add(1, 'd')
      .format('MMM Do, YYYY h:mm A');
    nagOn.simulate('change', nagOn);
    nagForm.find('form').simulate('submit');
    expect(nagCreate.mock.calls.length).toBe(1);
  });

  it('should call nagUpdate when all fields are valid', () => {
    const nagForm = mount(
      <Provider store={mockStore()}>
        <NagForm
          nagIndex={nagIndex}
          nagCreate={nagCreate}
          nagUpdate={nagUpdate}
          editNag={nagObject}
          editNagId={nagObject.id}
        />
      </Provider>
    );

    const nagTitle = nagForm.find('input#nag_title');
    nagTitle.instance().value = 'Nag Title';
    nagTitle.simulate('change', nagTitle);
    const nagOn = nagForm.find('input#nag_on');
    nagOn.instance().value = moment()
      .add(1, 'd')
      .format('MMM Do, YYYY h:mm A');
    nagOn.simulate('change', nagOn);
    nagForm.find('form').simulate('submit');
    expect(nagUpdate.mock.calls.length).toBe(1);
  });
});
