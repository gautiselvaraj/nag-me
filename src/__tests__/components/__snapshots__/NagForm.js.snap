// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NagForm /> should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 5px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  -webkit-flex: 0 0 25px;
  -ms-flex: 0 0 25px;
  flex: 0 0 25px;
  height: 15px;
  text-align: center;
}

.c2 {
  background-color: transparent;
  border: 0;
  cursor: pointer;
  display: inline-block;
  padding: 0;
}

.c2:hover {
  background-color: transparent;
}

.c14 {
  border: 1px solid props => props.theme.mainDarken;
  cursor: pointer;
  display: block;
  padding: 0 .75rem;
  width: 100%;
  box-sizing: border-box;
  font-size: 0.85rem;
  line-height: 2rem;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition-duration: 0.5s;
  transition-duration: 0.5s;
  -webkit-transition-property: background-color;
  transition-property: background-color;
}

.c3 {
  font-family: icons;
  font-size: 1rem;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  line-height: 1;
  speak: none;
  text-transform: none;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c3:before {
  content: "\\e900";
}

.c4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: center;
}

.c5 {
  font-size: 1.2rem;
  margin-top: 0.35rem;
  margin-bottom: 0.35rem;
}

.c6 {
  padding: 0.5rem;
}

.c7 {
  margin-bottom: 0.75rem;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 {
  font-size: 1rem;
  line-height: 2rem;
}

.c10 {
  border: 1px solid;
  border-radius: 5px;
  box-sizing: border-box;
  font-size: 1rem;
  height: 2rem;
  padding: 0 0.35rem;
  width: 100%;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 {
  border: 1px solid;
  box-sizing: border-box;
  font-size: 1rem;
  height: 2rem;
  margin-top: 0;
  margin-bottom: 0;
  width: 100%;
}

.c13 {
  margin-bottom: 0.75rem;
  margin-top: 1.5rem;
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <button
        className="c2"
        onClick={[Function]}
        title="Back"
      >
        <i
          className="c3"
        />
      </button>
    </div>
    <div
      className="c4"
    >
      <h3
        className="c5"
      >
        Create a nag
      </h3>
    </div>
    <div
      className="c1"
    />
  </div>
  <form
    className="c6"
    onSubmit={[Function]}
  >
    <div
      className="c7"
    >
      <div
        autoFocus={true}
        className="c8"
        name="title"
        onChange={[Function]}
        type="text"
        value=""
      >
        <label
          autoFocus={true}
          className="c9"
          htmlFor="nag_title"
          name="title"
          onChange={[Function]}
          type="text"
          value=""
        >
          What to nag about?
        </label>
        <input
          autoFocus={true}
          className="c10"
          id="nag_title"
          name="title"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="c7"
    >
      <div
        className="rdt"
      >
        <div>
          <div
            className="form-control c8"
            name="on"
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={true}
            type="text"
            value=""
          >
            <label
              className="form-control c9"
              htmlFor="nag_on"
              name="on"
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              readOnly={true}
              type="text"
              value=""
            >
              Nag from
            </label>
            <input
              className="form-control c10"
              id="nag_on"
              name="on"
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              readOnly={true}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="rdtPicker"
        >
          <div
            className="rdtDays"
          >
            <table>
              <thead>
                <tr>
                  <th
                    className="rdtPrev"
                    onClick={[Function]}
                  >
                    <span>
                      ‹
                    </span>
                  </th>
                  <th
                    className="rdtSwitch"
                    colSpan={5}
                    data-value={0}
                    onClick={[Function]}
                  >
                    January 2018
                  </th>
                  <th
                    className="rdtNext"
                    onClick={[Function]}
                  >
                    <span>
                      ›
                    </span>
                  </th>
                </tr>
                <tr>
                  <th
                    className="dow"
                  >
                    Su
                  </th>
                  <th
                    className="dow"
                  >
                    Mo
                  </th>
                  <th
                    className="dow"
                  >
                    Tu
                  </th>
                  <th
                    className="dow"
                  >
                    We
                  </th>
                  <th
                    className="dow"
                  >
                    Th
                  </th>
                  <th
                    className="dow"
                  >
                    Fr
                  </th>
                  <th
                    className="dow"
                  >
                    Sa
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td
                    className="rdtDay rdtOld rdtDisabled"
                    data-value={31}
                  >
                    31
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={1}
                  >
                    1
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={2}
                  >
                    2
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={3}
                  >
                    3
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={4}
                  >
                    4
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={5}
                  >
                    5
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={6}
                  >
                    6
                  </td>
                </tr>
                <tr>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={7}
                  >
                    7
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={8}
                  >
                    8
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={9}
                  >
                    9
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={10}
                  >
                    10
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={11}
                  >
                    11
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={12}
                  >
                    12
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={13}
                  >
                    13
                  </td>
                </tr>
                <tr>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={14}
                  >
                    14
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={15}
                  >
                    15
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={16}
                  >
                    16
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={17}
                  >
                    17
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={18}
                  >
                    18
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={19}
                  >
                    19
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={20}
                  >
                    20
                  </td>
                </tr>
                <tr>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={21}
                  >
                    21
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={22}
                  >
                    22
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={23}
                  >
                    23
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={24}
                  >
                    24
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={25}
                  >
                    25
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={26}
                  >
                    26
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={27}
                  >
                    27
                  </td>
                </tr>
                <tr>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={28}
                  >
                    28
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={29}
                  >
                    29
                  </td>
                  <td
                    className="rdtDay rdtToday rdtDisabled"
                    data-value={30}
                  >
                    30
                  </td>
                  <td
                    className="rdtDay rdtDisabled"
                    data-value={31}
                  >
                    31
                  </td>
                  <td
                    className="rdtDay rdtNew"
                    data-value={1}
                    onClick={[Function]}
                  >
                    1
                  </td>
                  <td
                    className="rdtDay rdtNew"
                    data-value={2}
                    onClick={[Function]}
                  >
                    2
                  </td>
                  <td
                    className="rdtDay rdtNew"
                    data-value={3}
                    onClick={[Function]}
                  >
                    3
                  </td>
                </tr>
                <tr>
                  <td
                    className="rdtDay rdtNew"
                    data-value={4}
                    onClick={[Function]}
                  >
                    4
                  </td>
                  <td
                    className="rdtDay rdtNew"
                    data-value={5}
                    onClick={[Function]}
                  >
                    5
                  </td>
                  <td
                    className="rdtDay rdtNew"
                    data-value={6}
                    onClick={[Function]}
                  >
                    6
                  </td>
                  <td
                    className="rdtDay rdtNew"
                    data-value={7}
                    onClick={[Function]}
                  >
                    7
                  </td>
                  <td
                    className="rdtDay rdtNew"
                    data-value={8}
                    onClick={[Function]}
                  >
                    8
                  </td>
                  <td
                    className="rdtDay rdtNew"
                    data-value={9}
                    onClick={[Function]}
                  >
                    9
                  </td>
                  <td
                    className="rdtDay rdtNew"
                    data-value={10}
                    onClick={[Function]}
                  >
                    10
                  </td>
                </tr>
              </tbody>
              <tfoot>
                <tr>
                  <td
                    className="rdtTimeToggle"
                    colSpan={7}
                    onClick={[Function]}
                  >
                    12:00 AM
                  </td>
                </tr>
              </tfoot>
            </table>
          </div>
        </div>
      </div>
    </div>
    <div
      className="c7"
    >
      <div
        className="c11"
        name="repeats"
        onChange={[Function]}
        value=""
      >
        <label
          className="c9"
          htmlFor="nag_repeats"
          name="repeats"
          onChange={[Function]}
          value=""
        >
          Repeat every
        </label>
        <select
          className="c12"
          id="nag_repeats"
          name="repeats"
          onChange={[Function]}
          value=""
        >
          <option
            value=""
          />
          <option
            value="1 min"
          >
            1
             
            min
            
          </option>
          <option
            value="2 min"
          >
            2
             
            min
            s
          </option>
          <option
            value="3 min"
          >
            3
             
            min
            s
          </option>
          <option
            value="5 min"
          >
            5
             
            min
            s
          </option>
          <option
            value="10 min"
          >
            10
             
            min
            s
          </option>
          <option
            value="15 min"
          >
            15
             
            min
            s
          </option>
          <option
            value="20 min"
          >
            20
             
            min
            s
          </option>
          <option
            value="30 min"
          >
            30
             
            min
            s
          </option>
          <option
            value="45 min"
          >
            45
             
            min
            s
          </option>
          <option
            value="1 hour"
          >
            1
             
            hour
            
          </option>
          <option
            value="2 hour"
          >
            2
             
            hour
            s
          </option>
          <option
            value="3 hour"
          >
            3
             
            hour
            s
          </option>
          <option
            value="4 hour"
          >
            4
             
            hour
            s
          </option>
          <option
            value="5 hour"
          >
            5
             
            hour
            s
          </option>
          <option
            value="6 hour"
          >
            6
             
            hour
            s
          </option>
          <option
            value="7 hour"
          >
            7
             
            hour
            s
          </option>
          <option
            value="8 hour"
          >
            8
             
            hour
            s
          </option>
          <option
            value="9 hour"
          >
            9
             
            hour
            s
          </option>
          <option
            value="10 hour"
          >
            10
             
            hour
            s
          </option>
          <option
            value="11 hour"
          >
            11
             
            hour
            s
          </option>
          <option
            value="12 hour"
          >
            12
             
            hour
            s
          </option>
          <option
            value="13 hour"
          >
            13
             
            hour
            s
          </option>
          <option
            value="14 hour"
          >
            14
             
            hour
            s
          </option>
          <option
            value="15 hour"
          >
            15
             
            hour
            s
          </option>
          <option
            value="16 hour"
          >
            16
             
            hour
            s
          </option>
          <option
            value="17 hour"
          >
            17
             
            hour
            s
          </option>
          <option
            value="18 hour"
          >
            18
             
            hour
            s
          </option>
          <option
            value="19 hour"
          >
            19
             
            hour
            s
          </option>
          <option
            value="20 hour"
          >
            20
             
            hour
            s
          </option>
          <option
            value="21 hour"
          >
            21
             
            hour
            s
          </option>
          <option
            value="22 hour"
          >
            22
             
            hour
            s
          </option>
          <option
            value="23 hour"
          >
            23
             
            hour
            s
          </option>
          <option
            value="1 day"
          >
            1
             
            day
            
          </option>
          <option
            value="2 day"
          >
            2
             
            day
            s
          </option>
          <option
            value="3 day"
          >
            3
             
            day
            s
          </option>
          <option
            value="4 day"
          >
            4
             
            day
            s
          </option>
          <option
            value="5 day"
          >
            5
             
            day
            s
          </option>
          <option
            value="6 day"
          >
            6
             
            day
            s
          </option>
          <option
            value="1 week"
          >
            1
             
            week
            
          </option>
          <option
            value="2 week"
          >
            2
             
            week
            s
          </option>
          <option
            value="3 week"
          >
            3
             
            week
            s
          </option>
          <option
            value="4 week"
          >
            4
             
            week
            s
          </option>
          <option
            value="5 week"
          >
            5
             
            week
            s
          </option>
          <option
            value="6 week"
          >
            6
             
            week
            s
          </option>
          <option
            value="7 week"
          >
            7
             
            week
            s
          </option>
          <option
            value="8 week"
          >
            8
             
            week
            s
          </option>
          <option
            value="9 week"
          >
            9
             
            week
            s
          </option>
          <option
            value="10 week"
          >
            10
             
            week
            s
          </option>
          <option
            value="1 month"
          >
            1
             
            month
            
          </option>
          <option
            value="2 month"
          >
            2
             
            month
            s
          </option>
          <option
            value="3 month"
          >
            3
             
            month
            s
          </option>
          <option
            value="4 month"
          >
            4
             
            month
            s
          </option>
          <option
            value="5 month"
          >
            5
             
            month
            s
          </option>
          <option
            value="6 month"
          >
            6
             
            month
            s
          </option>
          <option
            value="7 month"
          >
            7
             
            month
            s
          </option>
          <option
            value="8 month"
          >
            8
             
            month
            s
          </option>
          <option
            value="9 month"
          >
            9
             
            month
            s
          </option>
          <option
            value="10 month"
          >
            10
             
            month
            s
          </option>
          <option
            value="11 month"
          >
            11
             
            month
            s
          </option>
          <option
            value="1 year"
          >
            1
             
            year
            
          </option>
          <option
            value="2 year"
          >
            2
             
            year
            s
          </option>
          <option
            value="3 year"
          >
            3
             
            year
            s
          </option>
          <option
            value="4 year"
          >
            4
             
            year
            s
          </option>
          <option
            value="5 year"
          >
            5
             
            year
            s
          </option>
          <option
            value="6 year"
          >
            6
             
            year
            s
          </option>
          <option
            value="7 year"
          >
            7
             
            year
            s
          </option>
          <option
            value="8 year"
          >
            8
             
            year
            s
          </option>
          <option
            value="9 year"
          >
            9
             
            year
            s
          </option>
          <option
            value="10 year"
          >
            10
             
            year
            s
          </option>
        </select>
      </div>
    </div>
    <div
      className="c13"
    >
      <button
        className="c14"
        type="submit"
      >
        Start Nagging
      </button>
    </div>
  </form>
</div>
`;
